### PR TITLE
Prevent reusing stream ids

### PR DESCRIPTION
**Pull request by stephenc-pace**
When we receive a StreamReset frame (for eg; when a client terminates a stream on timeout), we currently cleanup the receive stream but leave the send stream.
Stream objects in nameko-grpc are just wrappers for a single underlying http/2 stream
```
        request_stream = SendStream(stream_id)
        response_stream = ReceiveStream(stream_id)
        self.receive_streams[stream_id] = response_stream
        self.send_streams[stream_id] = request_stream
```
So when we don't cleanup both of these objects on stream close we can attempt to send data again on the SendStream object, which will either attempt to send data on a closed stream `StreamClosedError` (if h2 hasn't discared its reference to the underyling stream yet, or if it has, h2 will create a new stream object with the previous stream_id, resulting in the `StreamIDTooLowError` error when we attempt to send data.

This PR fixes this by ensuring the SendStream object is removed from our references when we receive a StreamResetframe.

### CODE REVIEW

File: nameko_grpc/connection.py [hunk 1]
```diff
@@ -214,12 +214,15 @@ def stream_ended(self, event):
...
-        Close any `ReceiveStream` that was opened for this stream.
+        Close any Streams that we have opened for this stream_id
         """
         log.debug("stream reset, stream %s", event.stream_id)
         receive_stream = self.receive_streams.pop(event.stream_id, None)
         if receive_stream:
             receive_stream.close()
+        send_stream = self.send_streams.pop(event.stream_id, None)
+        if send_stream:
+            send_stream.close()
```
Comment (stephenc-pace) [inline nameko_grpc/connection.py:225]: Enusre our reference to the stream is closed

**Comment by stephenc-pace** [comment]
@mattbennett  Hope this makes sense. I need to add tests still to the PR (struggling with that part).

**Comment by stephenc-pace** [comment]
@mattbennett 
I haven't added tests as it's quite challenging to do so. 
We've had this in production for a number of months and it's resolved the issue identified without causing further problems so confident in it.

**Review by mattbennett — APPROVED** [review]
This looks like such a numbskull bug when it's on its own in a PR like this. Thanks for finding it though!

**Pull request merged**